docs/docs/exosphere/architecture.md

```
@@ -0,0 +1,150 @@
+- **State Manager**: Central orchestrator that manages state lifecycle and transitions
+- **Runtimes**: Distributed execution engines that process individual states
+- **Graph Templates**: Declarative workflow definitions
+- **States**: Individual execution units with inputs, outputs, and metadata
+
+## State Execution Model
+
+### State Lifecycle
+
+Each state in Exosphere follows a well-defined lifecycle:
+
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)